### PR TITLE
fix(fmt): Formatter formats invalid syntax into even worse invalid syntax

### PR DESCRIPTION
**Pull request by skartikey**
Issue - Formatter was formatting an invalid syntax script.

Fix - The fix is to check AST nodes(call ast.check()) and return if any error is found to the user, before proceeding to format the script. 

while fixing it we found that the AST walk doesn't actually traverse BuiltinStmt. So any errors that occur when parsing the type of builtin are just ignored. So to fix it AST walk for the BuiltinStmt has been added.

fixes: https://github.com/influxdata/vsflux/issues/242

### CODE REVIEW

File: libflux/flux-core/src/formatter/mod.rs [hunk 1]
```diff
@@ -26,6 +26,10 @@ pub fn convert_to_string(file: &File) -> Result<String> {
 /// ```
 pub fn format(contents: &str) -> Result<String> {
     let file = parse_string("".to_string(), contents);
+    let node = ast::walk::Node::File(&file);
+    if let Err(err) = ast::check::check(node) {
+        panic!("Script parsing failed. {:?}", err);
+    }
```
Comment (Marwes) [inline libflux/flux-core/src/formatter/mod.rs:32]:
```suggestion
    ast::check::check(node)?;
```

File: libflux/flux-core/src/formatter/tests.rs [hunk 2]
```diff
@@ -1268,3 +1268,17 @@ a",
     );
     assert_unchanged(r#"a = "a string literal with a tabstop '	'""#);
 }
+
+#[test]
+fn invalid_syntax() {
+    assert_unchanged(
```
Comment (Marwes) [inline libflux/flux-core/src/formatter/tests.rs:1274]: The `assert_unchanged` function will panic on errors. I think you should just call `format` here and check that you get an error.

**Review by Marwes — CHANGES_REQUESTED** [review]

**Comment by Marwes** [comment]
@skartikey Might be good to have someone else review this as one of the commits are mine now.

**Comment by skartikey** [comment]
@Marwes Just did that.

File: libflux/flux-core/src/formatter/tests.rs [hunk 3]
```diff
@@ -1231,7 +1231,7 @@ fn preserve_multiline_test() {
     return tables
         |> aggregateWindow(every: every, fn: fn, column: _column, createEmpty: true)
         |> fill(column: _column, value: defaultValue)
-        |> rename(fn: (column) => if column == _column then _as else column)"#,
+        |> rename(fn: (column) => if column == _column then _as else column)}"#,
```
Comment (rockstar) [inline libflux/flux-core/src/formatter/tests.rs:1234]: Was this a purposeful change? How did this ever work?

File: libflux/flux-core/src/formatter/tests.rs [hunk 4]
```diff
@@ -1268,3 +1268,17 @@ a",
...
+
+#[test]
+fn invalid_syntax() {
+    let script = r#"builtin diff : (
+        <-got: [A],
+        want: [A],
+        ?verbose: bool,
+        ?epsilon: float,
+        ?nansEqual: bool,
+        aoeustahoesih
```
Comment (rockstar) [inline libflux/flux-core/src/formatter/tests.rs:1280]: Haha! I know why this random string is here, but it's going to result is some really fun lore in the future. 😄

**Review by rockstar — APPROVED** [review]

File: libflux/flux-core/src/formatter/tests.rs (hunk 3 above)
Comment (skartikey) [reply libflux/flux-core/src/formatter/tests.rs:1234]: yeah, this was caught after my fix :)  Earlier there was no ast check at all, code was parsing the script without checking for error and calling the formatter on the parsed script. All fixed now.

**Review by Marwes — APPROVED** [review]

**Pull request merged**